src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -2455,8 +2455,7 @@ void InstanceKlass::metaspace_pointers_do(MetaspaceClosure* it) {
   // be rewritten during runtime (see Rewriter::rewrite_jsrs()). So setting the _methods to
   // be writable. The length check on the _methods is necessary because classes which
   // don't have any methods share the Universe::_the_empty_method_array which is in the RO region.
-  if (_methods != nullptr && _methods->length() > 0 &&
-      !can_be_verified_at_dumptime() && methods_contain_jsr_bytecode()) {
+  if (_methods != nullptr && _methods->length() > 0 && !can_be_verified_at_dumptime()) {
     // To handle jsr bytecode, new Method* maybe stored into _methods
```

Comment:
Based on the comment we only need to do this if the method does contain a JSR bytecode - now we will do it for all cases where verification can't be done at dumptime. ??

Reply:
Yes, we will put these in RW space the time.  Using the BytecodeStream to look for JSRs in unverified code is risky, so we avoid doing this optimization.

Reply:
So shouldn't the comment be updated to indicate that we cannot safely check if the jsr bytecode exists and so all methods that can't be verified are made writeable?